experimental/src/test/scala/org/testcontainers/testcontainers4s/containers/SingleContainerSuite.scala

```diff
@@ -0,0 +1,18 @@
+package org.testcontainers.testcontainers4s.containers
+
+import org.scalatest.FreeSpec
+import org.testcontainers.testcontainers4s.containers.scalatest.TestContainerForEach
+
+class SingleContainerSuite extends FreeSpec with TestContainerForEach {
+
+  override val containerDef = MySQLContainer.Def()
+
+  "foo" - {
+    "bar" in withContainers { db =>
+      assert(db.jdbcUrl.nonEmpty)
```

Comment:
https://github.com/testcontainers/testcontainers-scala/pull/59/files 
I'm thinking how its `def config: Config`could be used in this new schema. pass containers from each test explicitly?
It raises a question - how common is that pattern? How much additional changes will it introduce for current users?

Reply:
Not sure I fully understand your question. Config creation is absolutely unrelated to the testcontainers and could be done in the old and new API without significant changes.

Reply:
Config is referencing the container instance. Since there is no globally (in spec scope) accessible `container`"runtime" instance, that will stop working. or I'm missing something?

Reply:
Now I see. Yes, simple refactoring will be needed: instead of accessing spec scoped container user will have to create a function with `container` argument:
```scala
def config(container: JdbcContainer): Config = ??? // do anything you want
```
I don't think that this is the problem.